services/iam/app/resources/tenant_resource.rb

```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 class TenantResource < Iam::ApplicationResource
-  attributes :account_id, :alias, :name # :locale
+  attributes :account_id, :root_id, :alias, :name, :properties # :locale
```

Comment:
I'm not sure how did it worked without `root_id` before 🤔

Reply:
We don't need to set/unset the root id through the resource.

Reply:
But I've got an error when running the tests that `root` is expected. 
Let me double check that

Reply:
Yes, still betting that error in response, if I remove all `root` and `root_id` mentions:
```
{"errors"=>[{"title"=>"must exist", "detail"=>"root - must exist", "code"=>"100", "source"=>{"pointer"=>"/data/attributes/root"}, "status"=>"422"}]}
```

Reply:
ok. let me check

Reply:
Ok, After looking at the use case a few considerations:
1) There should be no way for us to create a tenant for now. A tenant has a one to one relationship with the root account. So creating a tenant would imply creating a root and therefore the logic should be reversed.
2) For this particular issue, we would be updating an existing tenant, so you don't need to expose the root_id.

For this issue, my suggestion is:
1. Disable the CREATE endpoint for tenant
2. Make the account_id and root_id fetchable but not settable 
3. For the request spec, instead of making a POST, you run a patch where you update the properties of an existing tenant.

I think this should solve your problem.

Reply:
Done